pkgs/tools/package-management/nix-prefetch-scripts/default.nix

```diff
@@ -1,40 +1,54 @@
+
+    buildInputs = [ makeWrapper ];
+
+    phases = [ "installPhase" "fixupPhase" ];
+    installPhase = ''
+      mkdir -p $out/bin
 
-  phases = [ "installPhase" "fixupPhase" ];
-  installPhase = ''
-    mkdir -p $out/bin
-    function copyScript {
-      local name=nix-prefetch-$1;
-      local src=$2;
       local wrapArgs=""
-      cp $src $out/bin/$name;
-      for dep in ''${@:3}; do
+      cp ${src} $out/bin/$name;
+      for dep in ${stdenv.lib.concatStringsSep " " deps}; do
         wrapArgs="$wrapArgs --prefix PATH : $dep/bin"
       done
       wrapArgs="$wrapArgs --prefix PATH : ${gnused}/bin"
       wrapArgs="$wrapArgs --set HOME : /homeless-shelter"
       wrapProgram $out/bin/$name $wrapArgs
-    }
-
-    copyScript "hg" ${../../../build-support/fetchhg/nix-prefetch-hg} ${mercurial}
-    copyScript "git" ${../../../build-support/fetchgit/nix-prefetch-git} ${git} ${coreutils}
-    copyScript "svn" ${../../../build-support/fetchsvn/nix-prefetch-svn} ${subversion}
-    copyScript "bzr" ${../../../build-support/fetchbzr/nix-prefetch-bzr} ${bazaar}
-    copyScript "cvs" ${../../../build-support/fetchcvs/nix-prefetch-cvs} ${cvs}
-    copyScript "zip" ${../../../build-support/fetchzip/nix-prefetch-zip} ${unzip} ${curl}
-  '';
-
-  meta = with stdenv.lib; {
-    description = "Collection of all the nix-prefetch-* scripts which may be used to obtain source hashes";
-    maintainers = with maintainers; [ bennofs ];
-    platforms = with stdenv.lib.platforms; unix;
-    # Quicker to build than to download, I hope
-    hydraPlatforms = [];
+    '';
+
+    meta = with stdenv.lib; {
+      description = "Script used to obtain source hashes for fetch${tool}";
+      maintainers = with maintainers; [ bennofs ];
+      platforms = with stdenv.lib.platforms; unix;
+      # Quicker to build than to download, I hope
+      hydraPlatforms = [];
+    };
+  };
+in rec {
+  nix-prefetch-bzr = mkPrefetchScript "bzr" ../../../build-support/fetchbzr/nix-prefetch-bzr [bazaar];
+  nix-prefetch-cvs = mkPrefetchScript "cvs" ../../../build-support/fetchcvs/nix-prefetch-cvs [cvs];
+  nix-prefetch-git = mkPrefetchScript "git" ../../../build-support/fetchgit/nix-prefetch-git [git coreutils];
+  nix-prefetch-hg  = mkPrefetchScript "hg"  ../../../build-support/fetchhg/nix-prefetch-hg   [mercurial];
+  nix-prefetch-svn = mkPrefetchScript "svn" ../../../build-support/fetchsvn/nix-prefetch-svn [subversion];
+  nix-prefetch-zip = mkPrefetchScript "zip" ../../../build-support/fetchzip/nix-prefetch-zip [unzip curl];
+
+  nix-prefetch-scripts = buildEnv {
+    name = "nix-prefetch-scripts";
+
+    paths = [ nix-prefetch-bzr nix-prefetch-cvs nix-prefetch-git nix-prefetch-hg nix-prefetch-svn nix-prefetch-zip ];
+
+    meta = with stdenv.lib; {
+      description = "Collection of all the nix-prefetch-* scripts which may be used to obtain source hashes";
+      maintainers = with maintainers; [ bennofs ];
+      platforms = with stdenv.lib.platforms; unix;
```

Comment:
I haven seen this `with platforms; platform;` thing a lot recently. Where does it come from? This is effectively the same as `platforms.platform`. I don't think we should use `with` when there is no real benefit as it slows down evaluation. In this case we don't even save key presses.